crates/starknet-types-core/src/felt/secret_felt.rs

```diff
@@ -0,0 +1,201 @@
+use crate::felt::{Felt, FromStrError};
+use lambdaworks_math::errors::ByteConversionError;
+use zeroize::Zeroize;
+
+/// A wrapper for a [Felt] that ensures the value is securely zeroized when dropped.
+///
+/// This type provides secure handling of sensitive [Felt] values (like private keys)
+/// by ensuring that the memory is properly cleared when the value is no longer needed.
+pub struct SecretFelt(Box<Felt>);
+
+impl zeroize::DefaultIsZeroes for Felt {}
+
+impl Zeroize for SecretFelt {
+    fn zeroize(&mut self) {
+        self.0.zeroize();
+    }
+}
+
+impl Drop for SecretFelt {
+    fn drop(&mut self) {
+        self.zeroize();
+    }
+}
+
+impl SecretFelt {
+    /// Creates a new [SecretFelt] from a [Felt] value and zeroize the original.
+    ///
+    /// It takes a mutable reference to a [Felt] value, creates a copy,
+    /// and then zeroize the original value to ensure it doesn't remain in memory.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use starknet_types_core::felt::{Felt, secret_felt::SecretFelt};
+    ///
+    /// let mut private_key = Felt::from_hex_unchecked("0x123...");
+    /// let secret_felt = SecretFelt::from_felt(&mut private_key);
+    /// // private_key is now zeroized (set to Felt::ZERO)
+    /// ```
+    pub fn from_felt(secret_scalar: &mut Felt) -> Self {
```

Comment:
maybe add a warning saying that this is only safe if the function is called immediately after the Felt was created. Because if the value is moved around before this call, it may have left some copy of its value here and there in memory